lib/repl_type_completor/type_analyzer.rb

```diff
@@ -925,8 +923,7 @@ def assign_parameters(node, scope, args, kwargs)
       node.posts.zip posts do |n, v|
         assign_required_parameter n, v, scope
       end
-      if node.rest&.name
-        # node.rest is Prism::RestParameterNode
+      if node.rest.is_a?(Prism::RestParameterNode) && node.rest.name
```

Comment:
There's a `if node.rest` condition a few lines above, which seems to already expect it to be `Prism::RestParameterNode`? Perhaps we can merge that and this together?

Reply:
I removed the confusing comment in the `if node.rest` condition and add a comment about `Prism::ImplicitRestNode` that does not have name.

The `if node.rest` above part is separating given args to reqs, posts, rest
`method do |a,b=1,*c| ` and given args `[Integer, String, Symbol]` to  `reqs = [Integer], opts = [String], posts = [], rest = [Symbol]`.

This part is assigning each of them.
```ruby
node.requireds.zip{assign}
node.optionals.zip{assign}
node.posts.zip{assign}
if node.rest; assign; end
```
I want to separate them, not to merge it together.

Reply:
I see, thanks for the clear explanation 👍